Fix various linting issues.

Mic92 opened the pull request.

alecbcs commented:
@dukzcry I couldn't push to your fork because I don't believe you allowed maintainers to push to this PR. 

Anyway I'd suggest instead of adding back the duplicate address parameter, you rewrite the option functions within the windows section like I have in the [windows-support branch](https://github.com/hyprspace/hyprspace/tree/windows-support). Feel free to copy that branch into this PR if it compiles for you.

dukzcry commented:
@alecbcs 

> I couldn't push to your fork because I don't believe you allowed maintainers to push to this PR.

Sorry, I don't see an option to change that :disappointed: 

> Anyway I'd suggest instead of adding back the duplicate address parameter, you rewrite the option functions within the windows section like I have in the windows-support branch. Feel free to copy that branch into this PR if it compiles for you.

Cool, thank you! It works just fine. I've moved back the interface enable thing from Up function back to interface creation stage though. Otherwise it fails with an error

alecbcs commented:
Great! No worries on moving the interface enable back to the main setup of the interface.

Mic92 committed:
remove unused vars/functions

Mic92 committed:
all constants need explicit types

See https://staticcheck.dev/docs/checks/#SA9004

Mic92 committed:
cli/up: add missing error checks

Mic92 committed:
server: fix potential race condition of chown when binding listening socket

a process waiting for the socket appear may bind before we can apply
filesystem permissions.

Mic92 commented on cli/up.go:
```diff
@@ -411,7 +419,12 @@ func streamHandler(stream network.Stream) {
+			stream.Close()
+			return
+		}
+		_, _ = tunDev.Iface.Write(packet[:size])
```
Is no error checks intended here?

max-privatevoid replied:
I'm fine with ignoring the error for now. I'm working on https://github.com/hyprspace/hyprspace/issues/16 right now, in which I'll clean up the error handling everywhere as well.

max-privatevoid commented:
Thanks
